src/xbot/common/command/BaseRobot.java

```diff
@@ -82,21 +88,37 @@ protected void initializeSystems() {
 
     @Override
     public void disabledInit() {
-        log.info("Disabled init");
+        log.info("Disabled init (" + getMatchContextString() + ")");
         propertyManager.saveOutAllProperties();
     }
 
     public void disabledPeriodic() {
         this.sharedPeriodic();
     }
+    
+    protected String getMatchContextString() {
+        DriverStation ds = DriverStation.getInstance();
+        return ds.getAlliance().toString() + ds.getLocation() + ", "
+            + ds.getMatchTime() + "s, "
+            + (ds.isDSAttached() ? "DS connected" : "DS disconnected") + ", "
+            + (ds.isFMSAttached() ? "FMS connected" : "FMS disconnected") + ", "
+            + "Is disabled: " + ds.isDisabled() + ", "
+            + "Is enabled: " + ds.isEnabled() + ", "
+            + "Is auto: " + ds.isAutonomous() + ", "
+            + "Is teleop: " + ds.isOperatorControl() + ", "
+            + "Is test: " + ds.isTest() + ", "
+            + "Is browned out: " + ds.isBrownedOut() + ", "
+            + "Is output enabled: " + ds.isSysActive() + ", "
+            + "Battery voltage: " + ds.getBatteryVoltage();
```

Comment:
I'm keeping it on one line right now so you can search through the file and you only need to see the single line with the match to get the info.